config/validation_test.go

```diff
@@ -0,0 +1,298 @@
+			},
+			"emptybackends": {
+				VolumeName:     "volume",
+				PolicyName:     "policy",
+				RuntimeOptions: RuntimeOptions{UseSnapshots: false},
+			},
+		},
+	}
+	PolicyConfigs = map[string]map[string]*Policy{
+		"valid": {
+			"basic": testPolicies["basic"],
+			"basicceph": {
+				Name: "basicceph",
+				Backends: &BackendDrivers{
+					CRUD:     "ceph",
+					Mount:    "ceph",
+					Snapshot: "ceph",
+				},
+			},
+			"basicnfs": {
+				Name: "basicnfs",
+				Backends: &BackendDrivers{
+					CRUD:     "",
+					Mount:    "nfs",
+					Snapshot: "",
+				},
+			},
+			"backendceph": {
+				Name:    "backendceph",
+				Backend: "ceph",
+			},
+			"backendnfs": {
+				Name:    "backendnfs",
+				Backend: "nfs",
+			},
+			"withbackendattr1": {
+				Name: "withbackendattr1",
+				Backends: &BackendDrivers{
+					CRUD:     "ceph",
+					Mount:    "ceph",
+					Snapshot: "ceph",
+				},
+				Backend: "nfs",
+			},
+			"withbackendattr2": {
+				Name: "withbackendattr2",
+				Backends: &BackendDrivers{
+					CRUD:     "",
+					Mount:    "nfs",
+					Snapshot: "",
+				},
+				Backend: "ceph",
+			},
+		},
+		"invalid": {
+			"emptyname": {
+				Name:    "",
+				Backend: "ceph",
+			},
+			"badbackend": {
+				Name:    "badbackend",
+				Backend: "bad",
+			},
+			"emptybackendconfig": {
+				Name: "emptybackendconfig",
+			},
+			"emptybackend": {
+				Name:    "emptybackend",
+				Backend: "", // omitempty
+			},
+			"emptybackendsconfig": {
+				Name: "emptybackendsconfig",
+				Backends: &BackendDrivers{ // this field will be omitted - "omitempty"
+					CRUD:     "",
+					Mount:    "",
+					Snapshot: "",
+				},
+			},
+			"badbackendsconfig": {
+				Name: "badbackendsconfig",
+				Backends: &BackendDrivers{
+					CRUD:     "bad",
+					Mount:    "bad",
+					Snapshot: "bad",
+				},
+			},
+			"nomount": {
+				Name:     "nomount",
+				Backends: &BackendDrivers{},
+			},
+			"nfsbackends": { //only "nfs" mount it is valid
+				Name: "crudnfs",
+				Backends: &BackendDrivers{
+					CRUD:     "nfs",
+					Mount:    "nfs",
+					Snapshot: "nfs",
+				},
+			},
+		},
+	}
+	RuntimeConfigs = map[string]map[string]*RuntimeOptions{
+		"valid": {
+			"snapshots": {
+				UseSnapshots: true,
+				Snapshot: SnapshotConfig{
+					Keep:      10,
+					Frequency: "1m",
+				},
+			},
+			"nosnapshots": {
+				UseSnapshots: false,
+			},
+		},
+		"invalid": {
+			"nosnapshotconfig": {
+				UseSnapshots: true, // requires snapshot configuration
+			},
+			"zerokeepvalue": {
+				UseSnapshots: true,
+				Snapshot: SnapshotConfig{
+					Keep:      0, // anything <1 is invalid
+					Frequency: "1m",
+				},
+			},
+			"emptyfrequency": {
+				UseSnapshots: true,
+				Snapshot: SnapshotConfig{
+					Keep:      1,
+					Frequency: "", // "" not allowed
+				},
+			},
+			"invalidfrequency": {
+				UseSnapshots: true,
+				Snapshot: SnapshotConfig{
+					Frequency: "badvalue", // accpets only (^[0-9]+.$)
```

Comment:
Erik, nothing sets `m`. go check just adds `^` & `$` 
https://github.com/yuva29/volplugin/blob/master/vendor/gopkg.in/check.v1/checkers.go#L307

So this will pass.